ADR/0039-spdx-support.md

```diff
@@ -0,0 +1,146 @@
+|                                 |  spdxElementId = <A-PACKAGE-ID>                            |
+|                                 |  relationshipType=BUILD_TOOL_OF                            |
+|                                 |  relatedSpdxElement=<ROOT-DOCUMENT-ID>                     |
+|                                 | }]                                                         |
+
+
+#### Metadata.tools
+The CycloneDX metadata.tools sub attributes that we are mostly interested in are the vendor and name elements. Information about the creation of the SPDX document can be stored into creationInfo. CreationInfo.creators element is basically a list of strings. There’s a vague specification about how it should be structured in the standard. Strings should be formatted in the following way: “<Attribute>: <Value>”. For example vendor should be stored as “Vendor: <vendor>”
+
+| SPDX Attribute                                 | CycloneDX Attribute                               |
+|------------------------------------------------|---------------------------------------------------|
+| Metadata.tools = [{“vendor”: “X”, “name”: “Y”] | CreationInfo.creators = [“Vendor: X”, “Tool: Y”]  |
+
+
+#### Merging SPDX
+##### Packages
+Packages of two SPDX documents can be merged together as a concatenation of two lists. In cycloneDX component elements can have only a single purl attribute, therefore component elements representing packages with the same name and version but with different purl have to be stored as multiple elements. CycloneDX package elements can bear multiple purls. Therefore multiple cycloneDX components can be squashed together into single SPDX package element with purls concatenated into a single list. Following rules are applied to packages merging process:
+- Packages with the same package name and versionInfo are squashed into single package element
+- Package with the same package name and versionInfo set to None (or empty) is squashed with package with the same name and non-empty versionInfo
+NOTE: packages cannot be merged together based on SPDXID attribute as there’s no specification in the spdx standard on how SPDXID should be calculated. Individual tools can calculate it differently while still passing condition to make it unique across the whole document.
+##### Relationships
+SPDX relationships represent graph/tree structure of relations of elements in the document. The Root element is the SPDX document itself. Individual packages are in relationship CONTAINS with the root document. In the case of packages which were used to build the container, packages are in relationship BUILD_TOOL_OF with the root document.
+NOTE: Packages can be wrapped in a “virtual package”. This package can have empty name and version and no attributes, or name can be set to directory or container name which was set as source for generating SBOM document. This element has insignificant information value and can be omitted.
```

Comment:
That's syft-specific, right?

Reply:
I haven't reviewed all spdx sbom generators in the world, so I can't really answer this question.

Reply:
Ack, @midnightercz could you call that out? Perhaps restructure the doc to separate out the handling of Syft specifically?

> I haven't reviewed all spdx sbom generators in the world, so I can't really answer this question.

Most of your assumptions are based on Syft and that's fine, let's just try to make it clear

Reply:
Or rather, let's separate "this is what Syft does" and "this is how our SBOMs should be", because what Syft does is often not a good example to follow